### PR TITLE
Add recipes for crafting input hatches

### PR DESCRIPTION
**Pull request by harrynull**
The tiering:
* Crafting Input Bus (item only): LuV gated by assline
* Crafting Input Slave: UV gated by UV sensor (neutronium?)
* Crafting Input Buffer: UIV/UMV gated by spacetime

Let me know if there are any suggestions / feedback

image
image
image
image
image

### CODE REVIEW

**Comment by harrynull** [comment]
Also @TimberDimber told me that they want a upgraded version with more pattern slots. I think depending on the tier of the base block, this can come either free, or with upgrade cards.

File: src/main/java/com/dreammaster/gthandler/recipes/AssemblingLineRecipes.java [hunk 1]
```diff
@@ -1,19 +1,19 @@
 package com.dreammaster.gthandler.recipes;
 
-import static gregtech.api.enums.Mods.GraviSuite;
-import static gregtech.api.util.GT_RecipeBuilder.HOURS;
-import static gregtech.api.util.GT_RecipeBuilder.MINUTES;
-import static gregtech.api.util.GT_RecipeBuilder.SECONDS;
-import static gregtech.api.util.GT_RecipeBuilder.WILDCARD;
+import static gregtech.api.enums.Mods.*;
```
Comment (BlueWeabo) [inline src/main/java/com/dreammaster/gthandler/recipes/AssemblingLineRecipes.java:3]: do not have wild imports.

File: src/main/java/com/dreammaster/gthandler/recipes/AssemblingLineRecipes.java [hunk 2]
```diff
@@ -132,6 +132,44 @@ public void run() {
                 .itemOutputs(ItemList.Energy_LapotronicOrb2.get(1L)).eut(TierEU.RECIPE_ZPM).duration(50 * SECONDS)
                 .addTo(AssemblyLine);
 
+        TT_recipeAdder.addResearchableAssemblylineRecipe(
```
Comment (BlueWeabo) [inline src/main/java/com/dreammaster/gthandler/recipes/AssemblingLineRecipes.java:135]: and an if (TecTech.isLoaded()) and then call TT_recipeAdder.

File: src/main/java/com/dreammaster/gthandler/recipes/AssemblingLineRecipes.java (hunk 1 above)
Comment (harrynull) [reply src/main/java/com/dreammaster/gthandler/recipes/AssemblingLineRecipes.java]: yea my IDE loves wildcard imports i guess

**Comment by Dream-Master** [comment]
its idea i have the same problem need to solve it manual now

**Comment by chochem** [comment]
in intellij its under settings->editor->codestyle->java->class count to use import with star: put some high number.

**Comment by AwonGardino** [comment]
May we not gate fluid one THAT far away? Mb just use living solder on that one?

**Review by TheEpicGamer274 — APPROVED** [review]
I think the recipe for the fluid one is gated quite nicely

**Review by meepmopper — CHANGES_REQUESTED** [review]
It should be more expensive (idk java at all and have no clue what the recipe is)

**Comment by Dream-Master** [comment]
> It should be more expensive (idk java at all and have no clue what the recipe is)

if you suggested a change you need to name it. You dont need to know java but you need to know GTNH very well

**Comment by meepmopper** [comment]
Sorry, I did not mean to submit that one. I think adding spatially enlarged fluid would be a good change as they are equivalent in tier and makes more sense thematically.

**Comment by Dream-Master** [comment]
@TheEpicGamer274  spatially enlarged fluid instead of space time ?

**Comment by TheEpicGamer274** [comment]
> @TheEpicGamer274 spatially enlarged fluid instead of space time ?

sure, sounds fine

**Comment by Dream-Master** [comment]
@harrynull can you cahnge that?

**Comment by GDCloudstrike** [comment]
there is an active discussion going on in github-discussion right now, please chime in if you have any suggestions

**Comment by dahgul333** [comment]
Ea! lets put liquid STARGATE! insted of spatially enlarged fluid! It will be more fun !(No .and spacetime tier fluid too)
gate it +- on Uhv- start Uev

**Comment by boubou19** [comment]
will probably need a follow up to fine tune the recipes, but overall tiering looks fine

**Review by boubou19 — APPROVED** [review]

**Pull request merged**